File: backend/src/main/kotlin/com/example/backend/models/AuditLog.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault
import java.time.Instant
import java.time.LocalDateTime

@Entity
@Table(name = "audit_logs")
class AuditLog {
@Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Long? = null

  @Column(name = "user_id")
  var userId: Long? = null

  @Column(name = "action", length = 50)
  var action: String? = null

  @Column(name = "method_name")
  var methodName: String? = null

  @Lob
  @Column(name = "arguments")
  var arguments: String? = null

  @Lob
  @Column(name = "result")
  var result: String? = null

  @Lob
  @Column(name = "exception")
  var exception: String? = null

  @ColumnDefault("CURRENT_TIMESTAMP")
  @Column(name = "timestamp")
  var timestamp: LocalDateTime? = null

  @Column(name = "created_date")
  var createdDate: LocalDateTime? = null

  @Column(name = "updated_date")
  var updatedDate: LocalDateTime? = null
}

File: backend_py/pharmaPython/app/models/audit_log.py
from sqlalchemy import Column, Integer, String, Text, DateTime
from sqlalchemy.sql import func
from app.core.db import Base

class AuditLog(Base):
    __tablename__ = "audit_logs"
    __table_args__ = {"extend_existing": True}

    id = Column(Integer, primary_key=True, autoincrement=True, nullable=False)
    user_id = Column(Integer, nullable=True, index=True)
    action = Column(String(50), nullable=True)
    method_name = Column(String(255), nullable=True)
    arguments = Column(Text, nullable=True)
    result = Column(Text, nullable=True)
    exception = Column(Text, nullable=True)
    timestamp = Column(DateTime, server_default=func.now(), nullable=True)
    created_date = Column(DateTime, nullable=True)
    updated_date = Column(DateTime, nullable=True)
